[SYNTH-18711] Update comment about sed syntax support (#28482)
DataDog/documentation: content/en/continuous_testing/environments/multiple_env.md

With this override, the URL `https://my-app.com/some/path` gets rewritten as `https://<deployment-prefix>.my-app.com/some/path`.
Notice that the URL path is not affected by the rewrite because it is not part of the substitution regex.

> 

Reviewer: Sorry @AntoineDona, I was only talking about the paragraph at line 83.

I know we could omit the fact that we support the sed-like syntax for simplicity, but this syntax still has the benefit of supporting regex modifiers, so I still think it's a good idea to mention it.

Here is my suggestion:
- Changed `s|<regex>|<rewriting rule>|<modifiers>` to be `s/<regex>/<rewriting rule>/<modifiers>`
- Said that the only benefit of the sed syntax is that it supports regex modifiers
- Promotes readability

```suggestion
<div class="alert alert-info">
Apart from the pipe <code>|</code> syntax presented above, <code>startUrlSubstitutionRegex</code> also supports the sed syntax: <code>s/&lt;regex&gt;/&lt;rewritting rule&gt;/&lt;modifiers&gt;</code>.</br></br>
But given this syntax uses a slash <code>/</code> separator, it may require escaping slashes from the URL which is error-prone. Unless you need regex modifiers, Datadog recommends using the pipe <code>|</code> syntax for better readability.
</div>
```

Notice that the URL path is not affected by the rewrite because it is not part of the substitution regex.

<div class="alert alert-info">
Apart from the pipe <code>|</code> syntax presented above, <code>startUrlSubstitutionRegex</code> also supports the sed syntax with modifiers: <code>s|&lt;regex&gt;|&lt;rewritting rule&gt;|&lt;modifiers&gt;</code>.</br></br>

> The sed syntax is often used with a slash <code>/</code> separator, for example: <code>s/&lt;regex&gt;/&lt;rewritting rule&gt;/&lt;modifier&gt;</code>. However, it can use any character as a delimiter. When working on a URL containing an abundant number of slashes, Datadog recommends using another character rather than escaping all slashes of the URL.

Reviewer: So basically, you just need to remove this paragraph

> Only the pipe <code>|</code> syntax is supported for URL substitution regexes. Sed syntax and other delimiter characters are not supported.

Reviewer: We do support a sed-like syntax `s/<matcher>/<replacer>/`. What's not supported is the fact that `sed` supports any character as a delimiter. You just have to stick to it the 3 times, e.g. `s$<matcher>$<replacer>$`

Notice that the URL path is not affected by the rewrite because it is not part of the substitution regex.

<div class="alert alert-info">
Apart from the pipe <code>|</code> syntax presented above, <code>startUrlSubstitutionRegex</code> also supports the sed syntax: <code>s/&lt;regex&gt;/&lt;rewriting rule&gt;/&lt;modifiers&gt;</code>.</br></br>

> But given this syntax uses a slash <code>/</code> separator, it may require escaping slashes from the URL which is error-prone. Unless you need regex modifiers, Datadog recommends using the pipe <code>|</code> syntax for better readability.

Reviewer 2: ```suggestion
Because sed syntax uses a slash <code>/</code> separator, it may require escaping slashes from the URL, which is error-prone. Unless you need regex modifiers, Datadog recommends using the pipe <code>|</code> syntax for better readability.
```

The <code>resourceUrlSubstitutionRegexes</code> is also applied to the first request, similarly to <code>startUrl</code> and <code>startUrlSubstitutionRegex</code>.
</div>

> <div class="alert alert-info">

Reviewer: same suggestion as above (don't forget to change `startUrlSubstitutionRegex` to `resourceUrlSubstitutionRegexes`)

<div class="alert alert-info">
The <code>resourceUrlSubstitutionRegexes</code> is also applied to the first request, similarly to <code>startUrl</code> and <code>startUrlSubstitutionRegex</code>.
</div>
<div class="alert alert-info">
Only the pipe <code>|</code> syntax is supported for URL substitution regexes. Sed syntax and other delimiter characters are not supported.
</div>

> ## Further reading

Reviewer 3: Not you but the `what's next` thing is broken and we should tell someone :D

Reviewer 2: @teodor2312 Can you tell me more about what's broken?

</div>

<div class="alert alert-info">
Apart from the pipe <code>|</code> syntax presented above, <code>resourceUrlSubstitutionRegexes</code> also supports the sed syntax: <code>s/&lt;regex&gt;/&lt;rewriting rule&gt;/&lt;modifiers&gt;</code>.</br></br>

> But given this syntax uses a slash <code>/</code> separator, it may require escaping slashes from the URL which is error-prone. Unless you need regex modifiers, Datadog recommends using the pipe <code>|</code> syntax for better readability.

Reviewer 2: ```suggestion
Because this syntax uses a slash <code>/</code> separator, it may require escaping slashes from the URL, which is error-prone. Unless you need regex modifiers, Datadog recommends using the pipe <code>|</code> syntax for better readability.
```
